File: go/src/auth/basic_auth.go
package auth

import (
	"encoding/base64"
	"net/http"
	"strings"
)

// BasicAuth implements the Authenticator interface for Basic Authentication
type BasicAuth struct {
	BaseAuth
	Username string
	Password string
}

// NewBasicAuth creates a new BasicAuth instance
func NewBasicAuth(username, password string) *BasicAuth {
	return &BasicAuth{
		BaseAuth: BaseAuth{
			Type: BasicAuth,
		},
		Username: username,
		Password: password,
	}
}

// Authenticate validates the Basic Authentication credentials
func (b *BasicAuth) Authenticate(r *http.Request) error {
	auth := r.Header.Get("Authorization")
	if auth == "" {
		return ErrMissingAuth
	}

	parts := strings.Split(auth, " ")
	if len(parts) != 2 || parts[0] != "Basic" {
		return ErrInvalidAuth
	}

	payload, err := base64.StdEncoding.DecodeString(parts[1])
	if err != nil {
		return ErrInvalidAuth
	}

	pair := strings.SplitN(string(payload), ":", 2)
	if len(pair) != 2 {
		return ErrInvalidAuth
	}

	if pair[0] != b.Username || pair[1] != b.Password {
		return ErrInvalidCredentials
	}

	return nil
}

// AddAuth adds Basic Authentication to the request
func (b *BasicAuth) AddAuth(r *http.Request) {
	auth := base64.StdEncoding.EncodeToString([]byte(b.Username + ":" + b.Password))
	r.Header.Set("Authorization", "Basic "+auth)
} 

File: php/src/Auth/BasicAuth.php
<?php

namespace Auth;

use Psr\Http\Message\ServerRequestInterface;

/**
 * BasicAuth implementation for HTTP Basic Authentication
 */
class BasicAuth extends BaseAuth
{
    private string $username;
    private string $password;

    /**
     * Create a new BasicAuth instance
     * @param string $username
     * @param string $password
     */
    public function __construct(string $username, string $password)
    {
        $this->type = AuthType::BASIC;
        $this->username = $username;
        $this->password = $password;
    }

    /**
     * Authenticate the request using Basic Authentication
     * @param ServerRequestInterface $request
     * @return bool
     */
    public function authenticate(ServerRequestInterface $request): bool
    {
        $auth = $request->getHeaderLine('Authorization');
        if (empty($auth)) {
            return false;
        }

        $parts = explode(' ', $auth);
        if (count($parts) !== 2 || $parts[0] !== 'Basic') {
            return false;
        }

        $decoded = base64_decode($parts[1]);
        if ($decoded === false) {
            return false;
        }

        $credentials = explode(':', $decoded);
        if (count($credentials) !== 2) {
            return false;
        }

        return $credentials[0] === $this->username && $credentials[1] === $this->password;
    }

    /**
     * Add Basic Authentication to the request
     * @param ServerRequestInterface $request
     * @return ServerRequestInterface
     */
    public function addAuth(ServerRequestInterface $request): ServerRequestInterface
    {
        $auth = base64_encode($this->username . ':' . $this->password);
        return $request->withHeader('Authorization', 'Basic ' . $auth);
    }
} 